### ISSUE TITLE
Extracting High and Low 32bit words from 64bit value

### ISSUE THREAD

**Issue by phanikishoreg** [gparmer/composite#217]
Add macros and clean-up code to extract High and Low 32bit words from 64bit value. 
Perhaps also to extract higher_n or lower_n bits! Which might be useful in extracting 16bit from 32bit values, and other more common bit manipulations as macros.

**Comment by gparmer** [comment]
Yes, making this generic is important for when we port to different platforms.  See the `round_*_to_*` functions for an example of this type of generality.

**Comment by gparmer** [comment]
Likely in `consts.h` along with the other pow-2 math.
